packages/frametricks/init.lua

```diff
@@ -153,12 +153,28 @@ function package:registerCommands ()
 
   self:registerCommand("showframe", function (options, _)
     local id = options.id or SILE.typesetter.frame.id
-    if id == "all" then
-      for _, frame in pairs(SILE.frames) do
-        SILE.outputter:debugFrame(frame)
+    options.allpages = options.allpages or true
+
+    local show = function()
+      if id == "all" then
+        for _, frame in pairs(SILE.frames) do
+          SILE.outputter:debugFrame(frame)
+        end
+      else
+        SILE.outputter:debugFrame(SILE.getFrame(id))
+      end
+    end
+
+    if options.allpages and options.allpages ~= "false" then
+      show()
+      local oldNewPage = SILE.documentState.documentClass.newPage
+      SILE.documentState.documentClass.newPage = function (self_)
+        local page = oldNewPage(self_)
+        show()
+        return page
       end
```

Comment:
```suggestion
      self.class:registerHook("newpage", show)
```

Reply:
Thanks for the tip. I got the following error when testing:

`Using code at: ./core/frame.lua:261: Passed a table, expected a string

So when I do ` self.class:registerHook("newpage", show(id)` the error isn't raised, but it still gives the same output: just the first page is debugged...